src/main/scala/org/scalameter/BasePerformanceTest.scala

```diff
@@ -2,13 +2,17 @@ package org.scalameter
 
 
 import java.util.Date
+import org.scalameter.picklers.Pickler
 import org.scalameter.utils.Tree
+import scala.language.existentials
 import scala.util.DynamicVariable
 
 
 
 abstract class BasePerformanceTest {
 
+  type Result
```

Comment:
We have specific `Bench` subclasses that assume a specific type like `Double`. For those, we can specify the default `measurer`, `warmer`, `reporter`, etc.

If there is a non-`Double` measurer, such as the method invocation measurer, then it should have its own `Bench` subclass with default values. And we can only provide it after we have the right reporter, anyway.